lib/_http_common.js

```diff
@@ -43,22 +23,11 @@ function parserOnHeadersComplete(versionMajor, versionMinor, headers, method,
                                  url, statusCode, statusMessage, upgrade,
                                  shouldKeepAlive) {
   var parser = this;
-
-  if (!headers) {
-    headers = parser._headers;
-    parser._headers = [];
-  }
-
-  if (!url) {
-    url = parser._url;
-    parser._url = '';
-  }
-
-  parser.incoming = new IncomingMessage(parser.socket);
-  parser.incoming.httpVersionMajor = versionMajor;
-  parser.incoming.httpVersionMinor = versionMinor;
-  parser.incoming.httpVersion = versionMajor + '.' + versionMinor;
-  parser.incoming.url = url;
+  var stream = parser.incoming = new IncomingMessage(parser.socket);
```

Comment:
If it sounds weird to you, it sounds weird to me too, so I'd like some confirmation before you act on that though :)